### DISCUSSION TITLE
Changing the default storage provider?

### DISCUSSION THREAD

**Discussion by damacus** [k3s-io/k3s#3445] [Q&A] [answered]
**Environmental Info:**
K3s Version:  `v1.21.0+k3s1`

Node(s) CPU architecture, OS, and Version: 

5.8.0-1021-raspi #24-Ubuntu SMP PREEMPT Tue Apr 13 11:00:57 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux

Cluster Configuration:

3 servers in HA

**Describe the bug:**

Local-path storage-class always reverts to being the default provisioner. 

**Steps To Reproduce:**

- Add a second storage-class (in my case the NFS client provisoner)
- Set that storage-class as the default
- Restart a node
- Local-path storage-class is now *also* the default storage class, giving us 2 default storage classes

**Expected behavior:**

Local-path should not be the default storage on reboot

**Actual behavior:**

Two local storage classes, the new storage class and the defualt shipped class.

**Comment by ChristianCiach** [comment, marked as answer]
The source of truth seems to be `/var/lib/rancher/k3s/server/manifests/local-storage.yaml` (https://github.com/k3s-io/k3s/blob/02a5bee62f5d7a0811a6a9401fc9ff5b1399c0bc/manifests/local-storage.yaml#L91) which is re-applied on every reboot of k3s. Looks like you can change the default there.

Unfortunately, this component is not deployed as a helm chart. Otherwise I would point you to the documentation of `HelmChartConfig`: https://rancher.com/docs/k3s/latest/en/helm/#customizing-packaged-components-with-helmchartconfig

A better option may be to install K3s using `--disable local-storage`. You can then install the local-storage provider by using a Helm chart if you still need it: https://github.com/rancher/local-path-provisioner/blob/master/deploy/chart/README.md#installing-the-chart

**Reply by damacus** [reply]
I've gone with this method for now. It's not the nicest, but it definitely makes it so I don't have to patch the default storage class on every boot

```shell
kubectl patch storageclass local-path -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"false"}}}'
shell
```

**Reply by vincentertainment** [reply]
I'm having the same issue. I've tried editing local-storage.yaml, I've tried the patch mentioned above, even deleting the local-path storage class. It keeps coming back on reboot as the default storage class. Not sure if this is a factor, but I'm managing the K3s cluster with Rancher.

**Comment by brandond** [comment]
Yes, `--disable local-storage` is the best way to do this. If you want to still use local-storage, you can copy and customize `/var/lib/rancher/k3s/server/manifests/local-storage.yaml` into something with a unique name before disabling the built-in local-storage component.

**Reply by fungiboletus** [reply]
It would be great if it was possible to be able to use another default storage class without disabling the builtin one.

**Comment by GreyXor** [comment]
FYI, you can use `default-local-storage-path` config to change and persist the default path of local-storage
